### PR TITLE
Check the whole workspace when running clippy

### LINKED ISSUE #1533: [Feature] CI: remove -p argument from `cargo clippy -p risc0-zkvm` for wider clippy coverage
## Feature
We should drop the `-p risc0-zkvm` from the clippy command in main.yml. -p seems to limit clippy coverage to risc0-zkvm only and removing this flag results in clippy warnings in risc0-zkp. 

## Motivation
We want to make sure we're running clippy on all of the crates

## Implementation
1. Wait until Frank finishes the parallel witgen work because it touches files that clippy complains about.
2. Drop the -p risc0-zkvm flag
3. Fix any clippy warnings that appear.

### PR DESCRIPTION
**Pull request by Cardosaum**
Closes #1533

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -138,7 +138,7 @@ jobs:
       - run: cargo fmt --all --check --manifest-path bonsai/Cargo.toml
       - run: cargo fmt --all --check --manifest-path examples/Cargo.toml
       - run: cargo fmt --all --check --manifest-path tools/crates-validator/Cargo.toml
-      - run: cargo clippy -p risc0-zkvm
+      - run: cargo clippy --workspace --exclude doc-test --exclude fibonacci-methods --exclude risc0-zkvm-methods --exclude xtask
```
Comment (flaub) [inline .github/workflows/main.yml:141]: Why do we need exclusions?

Comment (Cardosaum) [reply .github/workflows/main.yml:141]: For the listed packages above I'm getting compilation error like the following:

```
risc0-zkvm-methods-guest:    Compiling risc0-zkvm-methods-guest v0.1.0 (/home/matheus/git/github/risc0/risc0/risc0/zkvm/methods/guest)                                                                            
risc0-zkvm-methods-guest: error[E0463]: can't find crate for `core`
risc0-zkvm-methods-guest:   |
risc0-zkvm-methods-guest:   = note: the `riscv32im-risc0-zkvm-elf` target may not be installed
risc0-zkvm-methods-guest:   = help: consider downloading the target with `rustup target add riscv32im-risc0-zkvm-elf`
risc0-zkvm-methods-guest:   = help: consider building the standard library from source with `cargo build -Zbuild-std`
risc0-zkvm-methods-guest: 
risc0-zkvm-methods-guest: For more information about this error, try `rustc --explain E0463`.
risc0-zkvm-methods-guest: error: could not compile `risc0-zkvm-methods-guest` (bin "slice_io") due to previous error
risc0-zkvm-methods-guest: warning: build failed, waiting for other jobs to finish...
risc0-zkvm-methods-guest: error: could not compile `risc0-zkvm-methods-guest` (bin "hello_commit") due to previous error
risc0-zkvm-methods-guest: error: could not compile `risc0-zkvm-methods-guest` (bin "multi_test") due to previous error
error: failed to run custom build command for `risc0-zkvm-methods v0.22.0-alpha.1 (/home/matheus/git/github/risc0/risc0/risc0/zkvm/methods)`
```

Maybe there's something I should do on my local dev environment in order to provide `riscv32im-risc0-zkvm-elf` support? 

I'd imagine I already have it though (?)

Comment (Cardosaum) [reply .github/workflows/main.yml:141]: I'm not sure why `clippy` fails though as if I run `cargo check` everything compiles correctly.

**Review by flaub — APPROVED** [review]

**Comment by Cardosaum** [comment]
Closed in favor of #1706. We'll go with the approach of adding clippy coverage to each crate individually

**Pull request closed without merging**